std/meta.zig

```diff
@@ -115,6 +115,66 @@ test "std.meta.Child" {
     testing.expect(Child(?u8) == u8);
 }
 
+/// Given an array/pointer type, return the slice type `[]Child`.
+/// Preserves all pointer attributes such as `const`/`volatile` etc.
+pub fn SliceType(comptime T: type) type {
+    return switch (@typeInfo(T)) {
+        TypeId.Array => |info| @compileError("SliceType not implemented for arrays"),
+        TypeId.Pointer => |info| @Type(builtin.TypeInfo { .Pointer = builtin.TypeInfo.Pointer {
+            .size = builtin.TypeInfo.Pointer.Size.Slice,
+            .is_const = info.is_const,
+            .is_volatile = info.is_volatile,
+            .alignment = info.alignment,
+            .child = info.child,
+            .is_allowzero = info.is_allowzero,
+        }}),
+        else => @compileError("Expected pointer or array type, " ++ "found '" ++ @typeName(T) ++ "'"),
+    };
+}
+
+test "std.meta.SliceType" {
+    testing.expect(SliceType([]u8) == []u8);
```

Comment:
Not sure how much testing I should do here...

Reply:
Basic functionality is fine.